federation/pkg/kubefed/util/util.go

```diff
@@ -210,3 +210,26 @@ func WaitForPods(clientset *client.Clientset, pods []string, namespace string) e
 	})
 	return err
 }
+
+func GetFirstNodeIP(clientset *client.Clientset) ([]string, error) {
```

Comment:
This function seems like something that might exist elsewhere in the codebase.  Have you looked for and not found an existing implementation?

If this function is required, please ensure it is tested.  It may be helpful to break the function apart to avoid having to mock the client (i.e. focus on testing a function that accepts a NodeList and returns the first address).